False positives `unsupported-assignment-operation` and `unsubscriptable-object` for `TypedDict`s

### Steps to reproduce

Given a file `test.py`:
```python
# pylint: disable=missing-docstring, pointless-statement
from typing import TypedDict


class MyDict(TypedDict):
    sample: int


my_dict = MyDict(sample=1)

my_dict['sample']
my_dict['sample'] = 2

```

### Current behavior

Result of `pylint test.py`:
```
************* Module test
test.py:11:0: E1136: Value 'my_dict' is unsubscriptable (unsubscriptable-object)
test.py:12:0: E1137: 'my_dict' does not support item assignment (unsupported-assignment-operation)
```

### Expected behavior

It should be passing.

### pylint --version output

Result of `pylint --version` output:

```
pylint 3.0.0a3
astroid 2.6.0
Python 3.9.5 (default, May 24 2021, 12:50:35) 
[GCC 11.1.0]

```

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -8,7 +8,7 @@ on:
   pull_request: ~
 
 env:
-  CACHE_VERSION: 2
+  CACHE_VERSION: 3
   DEFAULT_PYTHON: 3.6
   PRE_COMMIT_CACHE: ~/.cache/pre-commit
 

--- a/requirements_test_min.txt
+++ b/requirements_test_min.txt
@@ -1,4 +1,4 @@
 -e .
-astroid==2.6.0  # Pinned to a specific version for tests
+astroid==2.6.1  # Pinned to a specific version for tests
 pytest~=6.2
 pytest-benchmark~=3.4

--- a/setup.cfg
+++ b/setup.cfg
@@ -41,7 +41,7 @@ project_urls =
 [options]
 packages = find:
 install_requires =
-    astroid>=2.6.0,<2.7 # (You should also upgrade requirements_test_min.txt)
+    astroid>=2.6.1,<2.7 # (You should also upgrade requirements_test_min.txt)
     isort>=4.2.5,<6
     mccabe>=0.6,<0.7
     toml>=0.7.1

--- a/tests/functional/t/typedDict.py
+++ b/tests/functional/t/typedDict.py
@@ -1,5 +1,5 @@
 """Test typing.TypedDict"""
-# pylint: disable=invalid-name,missing-class-docstring
+# pylint: disable=invalid-name,missing-class-docstring,pointless-statement
 import typing
 from typing import TypedDict
 
@@ -7,12 +7,19 @@ from typing import TypedDict
 class CustomTD(TypedDict):
     var: int
 
+
 class CustomTD2(typing.TypedDict, total=False):
     var2: str
+
 
 class CustomTD3(CustomTD2):
     var3: int
 
+
 CustomTD4 = TypedDict("CustomTD4", var4=bool)
 
 CustomTD5 = TypedDict("CustomTD5", {"var5": bool})
+
+my_dict = CustomTD(var=1)
+my_dict["var"]
+my_dict["var"] = 2

